snuba/snuba_migrations/uptime_monitor_checks/0001_uptime_monitor_checks.py

```diff
@@ -0,0 +1,80 @@
+from typing import List, Sequence
+
+from snuba.clickhouse.columns import UUID, Column, String, UInt
+from snuba.clusters.storage_sets import StorageSetKey
+from snuba.migrations import migration, operations, table_engines
+from snuba.migrations.columns import MigrationModifiers as Modifiers
+from snuba.migrations.operations import OperationTarget, SqlOperation
+from snuba.utils.schemas import DateTime64
+
+storage_set = StorageSetKey.UPTIME_MONITOR_CHECKS
+table_prefix = "uptime_monitor_checks"
+local_table_name = f"{table_prefix}_local"
+dist_table_name = f"{table_prefix}_dist"
+
+## what about all the fancy codecs? do we need those?
+columns: List[Column[Modifiers]] = [
+    Column("organization_id", UInt(64)),
+    Column("project_id", UInt(64)),
+    Column("environment", String(Modifiers(nullable=True, low_cardinality=True))),
+    Column("uptime_subscription_id", UUID()),
+    Column("uptime_check_id", UUID()),
+    Column("scheduled_check_time", DateTime64(3)),  # millisecond precision
+    Column("timestamp", DateTime64(3)),  # millisecond precision
+    Column("duration_ms", UInt(64)),
+    Column("region_slug", String(Modifiers(low_cardinality=True))),
+    Column("check_status", String(Modifiers(low_cardinality=True))),
+    Column(
+        "check_status_reason",
+        String(Modifiers(nullable=True, low_cardinality=True)),
+    ),
+    Column("http_status_code", UInt(16)),
+    Column("trace_id", UUID()),
+    Column("retention_days", UInt(16)),
+]
+
+
+class Migration(migration.ClickhouseNodeMigration):
+    blocking = False
+
+    def forwards_ops(self) -> Sequence[SqlOperation]:
+        return [
+            operations.CreateTable(
+                storage_set=storage_set,
+                table_name=local_table_name,
+                columns=columns,
+                engine=table_engines.ReplacingMergeTree(
+                    primary_key="(organization_id, project_id, toDateTime(timestamp), uptime_check_id, trace_id)",
+                    order_by="(organization_id, project_id, toDateTime(timestamp), uptime_check_id, trace_id)",
```

Comment:
Is there only 1 check per trace? You might something more specific here otherwise the ReplacingMergeTree will merge 2 rows with the same sort key.
```
The engine differs from MergeTree in that it removes duplicate entries with the same sorting key value (ORDER BY table section, not PRIMARY KEY).
```
https://clickhouse.com/docs/en/engines/table-engines/mergetree-family/replacingmergetree

If you're not going to update values or if you don't care about duplicates, you could just use a MergeTree.

Reply:
only one check per trace. duplicates would not be ideal